server/src/main/java/org/elasticsearch/search/fetch/PreloadedFieldLookupProvider.java

```diff
@@ -26,14 +29,21 @@
  */
 class PreloadedFieldLookupProvider implements LeafFieldLookupProvider {
 
-    Map<String, List<Object>> storedFields;
-    LeafFieldLookupProvider backUpLoader;
-    Supplier<LeafFieldLookupProvider> loaderSupplier;
+    private Set<String> preloadedStoredFields;
+    private Map<String, List<Object>> storedFields;
+    private String id;
+    private LeafFieldLookupProvider backUpLoader;
+    private Supplier<LeafFieldLookupProvider> loaderSupplier;
 
     @Override
     public void populateFieldLookup(FieldLookup fieldLookup, int doc) throws IOException {
         String field = fieldLookup.fieldType().name();
-        if (storedFields.containsKey(field)) {
+
+        if (field.equals(IdFieldMapper.NAME)) {
```

Comment:
_id was previously always loaded via the backup loader when requested via script or via fetch fields. That causes overhead as the field is already available at all times, no need to go and fetch it from stored fields!